aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -145,14 +145,28 @@ TORCH_IMPL_FUNC(func_out) (const Tensor& self, const Tensor& result) {  \
   func_stub(device_type(), *this);                                      \
 }
 
+// This macro is as optional as the one above. torch.(ceil|floor|round|trunc) are no-ops for integers
+// See gh-70918
+#define CREATE_UNARY_TORCH_IMPL_INTEGER_NO_OP_FUNC(func_out, func_stub)                                \
+TORCH_IMPL_FUNC(func_out) (const Tensor& self, const Tensor& result) {  \
+  if (self.dtype() == at::kLong) {                                      \
```

Comment:
I think you might want to use the following helper function.
https://github.com/pytorch/pytorch/blob/089203f8bc71c64688def521284e32aee3fb5989/aten/src/ATen/native/UnaryOps.cpp#L321

Reply:
Also you'd have to update the corresponding OpInfo entries with the new dtype support.